core/Tracker/Request.php

```diff
@@ -738,21 +739,26 @@ public function getVisitorId()
             }
         }
 
-        // - If set to use 3rd party cookies for Visit ID, read the cookie
-        if (!$found) {
-            $useThirdPartyCookie = $this->shouldUseThirdPartyCookie();
-            if ($useThirdPartyCookie) {
-                $idVisitor = $this->getThirdPartyCookieVisitorId();
-                if(!empty($idVisitor)) {
-                    $found = true;
+        $cache = Tracker\Cache::getCacheGeneral();
+
+        // Only check for cookie values if cookieless tracking is NOT forced
+        if (!isset($cache['forceCookielessTracking']) || !$cache['forceCookielessTracking']) {
```

Comment:
btw here it be good if we used some constants as it's set in a different place and read here

Reply:
will change that to directly use the privacy config (which will use tracker cache)